packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart

```diff
@@ -144,96 +90,86 @@ void main() {
-          'upgrade',
-          '--continue',
-          '--no-version-check',
-        ],
-        environment: anyNamed('environment'),
-        workingDirectory: anyNamed('workingDirectory'),
-      ));
       expect(testLogger.statusText, contains('Flutter is already up to date'));
+      expect(processManager.hasRemainingExpectations, isFalse);
     }, overrides: <Type, Generator>{
       ProcessManager: () => processManager,
       Platform: () => fakePlatform,
     });
 
     testUsingContext('fetchRemoteRevision returns revision if git succeeds', () async {
       const String revision = 'abc123';
-      when(processManager.run(
-        <String>['git', 'fetch', '--tags'],
-        environment:anyNamed('environment'),
-        workingDirectory: anyNamed('workingDirectory')),
-      ).thenAnswer((Invocation invocation) async {
-        return FakeProcessResult()
-          ..exitCode = 0;
-      });
-      when(processManager.run(
-        <String>['git', 'rev-parse', '--verify', '@{u}'],
-        environment:anyNamed('environment'),
-        workingDirectory: anyNamed('workingDirectory')),
-      ).thenAnswer((Invocation invocation) async {
-        return FakeProcessResult()
-          ..exitCode = 0
-          ..stdout = revision;
-      });
+
+      processManager.addCommands(<FakeCommand>[
+        const FakeCommand(command: <String>[
+          'git', 'fetch', '--tags'
+        ]),
+        const FakeCommand(command: <String>[
+          'git', 'rev-parse', '--verify', '@{u}',
+        ],
+        stdout: revision),
+      ]);
+
       expect(await realCommandRunner.fetchRemoteRevision(), revision);
+      expect(processManager.hasRemainingExpectations, isFalse);
     }, overrides: <Type, Generator>{
       ProcessManager: () => processManager,
       Platform: () => fakePlatform,
     });
 
     testUsingContext('fetchRemoteRevision throws toolExit if HEAD is detached', () async {
-      when(processManager.run(
-        <String>['git', 'fetch', '--tags'],
-        environment:anyNamed('environment'),
-        workingDirectory: anyNamed('workingDirectory')),
-      ).thenAnswer((Invocation invocation) async {
-        return FakeProcessResult()..exitCode = 0;
-      });
-      when(processManager.run(
-        <String>['git', 'rev-parse', '--verify', '@{u}'],
-        environment:anyNamed('environment'),
-        workingDirectory: anyNamed('workingDirectory')),
-      ).thenThrow(const ProcessException(
-        'git',
-        <String>['rev-parse', '--verify', '@{u}'],
-        'fatal: HEAD does not point to a branch',
-      ));
-      expect(
-        () async => await realCommandRunner.fetchRemoteRevision(),
+      processManager.addCommands(<FakeCommand>[
+        const FakeCommand(command: <String>[
+          'git', 'fetch', '--tags'
+        ]),
+        FakeCommand(command: const <String>[
+          'git', 'rev-parse', '--verify', '@{u}',
+        ],
+            onRun: () {
```

Comment:
formatting is kind of weird here, could you align the FakeCommand args?

Reply:
Done.